python-checks/src/main/java/org/sonar/python/checks/cdk/AbstractIamPolicyStatementCheck.java

```diff
@@ -28,54 +28,45 @@
 import org.sonar.plugins.python.api.tree.CallExpression;
 import org.sonar.plugins.python.api.tree.DictionaryLiteral;
 import org.sonar.plugins.python.api.tree.Expression;
-import org.sonar.plugins.python.api.tree.ListLiteral;
 
 import static org.sonar.python.checks.cdk.CdkPredicate.isFqn;
-import static org.sonar.python.checks.cdk.CdkPredicate.isString;
 
-public class CdkIamUtils {
+public abstract class AbstractIamPolicyStatementCheck extends AbstractCdkResourceCheck {
```

Comment:
The direction of changes and using abstract class looks nice, especially the rule implementation - small clean code. However:
1. Have you considered using the Util class instead of the abstract? (Delegation over inheritance)
2. Currently this class mix protected and public static methods, so I'm a little bit confused. Is it still Util like class or the class I should extend?

It's not a pushback from my side, I'm just challenging the current shape and usage of the class.